src/radar/radar/toyota_radar_controller.py

```diff
@@ -0,0 +1,124 @@
+    (0x160, ECU.DSU, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA), 1, 7,
+     '\x00\x00\x08\x12\x01\x31\x9c\x51'),
+    (0x161, ECU.DSU, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA), 1, 7,
+     '\x00\x1e\x00\x00\x00\x80\x07'),
+    (0x365, ECU.DSU, (CAR.RAV4, CAR.COROLLA), 0, 20, '\x00\x00\x00\x80\xfc\x00\x08'),
+    (0x366, ECU.DSU, (CAR.RAV4, CAR.COROLLA), 0, 20, '\x00\x72\x07\xff\x09\xfe\x00'),
+    (0x4CB, ECU.DSU, (CAR.PRIUS, CAR.RAV4H, CAR.LEXUS_RXH, CAR.RAV4, CAR.COROLLA), 0, 100,
+     '\x0c\x00\x00\x00\x00\x00\x00\x00'),
+]
+
+
+class ToyotaRadarController(Node):
+    """
+    This radar controller is hardcoded to work only with the Toyota Corolla/Rav4/Camry 2017 Denso unit
+    """
+
+    def __init__(self):
+        super().__init__('radar')
+        # self.tracks_pub = self.create_publisher(String, 'radar_tracks')
+        self.can_pub = self.create_publisher(CanMessage, 'can_send')
+        self.adas_db = cantools.db.load_file(os.path.join(opendbc.DBC_PATH, 'toyota_prius_2017_adas.dbc'))
+        self.db = cantools.db.load_file(os.path.join(opendbc.DBC_PATH, 'toyota_prius_2017_pt_generated.dbc'))
+
+        self.can_sub = self.create_subscription(CanMessage, 'can_recv', self.on_can_message)
+        self.radar_pub = self.create_publisher(RadarTracks, 'radar_tracks')
+        self.power_on_timer = self.create_timer(1.0/100, self.power_on_radar)
+        self.radar_is_on = False
+        self.frame = 0
+        self.last_update_ms = util.ms_since_epoch()
+        self.current_radar_counter = 0
+        self.radar_tracks_msg = RadarTracks()
+
+    def power_on_radar(self):
+        acc_message = self.db.get_message_by_name('ACC_CONTROL')
+        acc_msg = acc_message.encode(
+            {"ACCEL_CMD": 0.0, "SET_ME_X63": 0x63, "SET_ME_1": 1, "RELEASE_STANDSTILL": 1, "CANCEL_REQ": 0,
+             "CHECKSUM": 113})
+        msg = CanMessage(id=acc_message.frame_id, interface=CanMessage.CANTYPE_RADAR, data=acc_msg)
+        # print("sending control command {} on bus: {}, vl: {}".format(acc_message.frame_id, 0, acc_msg))
+        self.can_pub.publish(msg)
+
+        for (addr, ecu, cars, bus, fr_step, vl) in STATIC_MSGS:
+            if self.frame % fr_step == 0:
+                if addr in (0x489, 0x48a) and bus == 0:
+                    # add counter for those 2 messages (last 4 bits)
+                    cnt = ((self.frame / 100) % 0xf) + 1
+                    if addr == 0x48a:
+                        # 0x48a has a 8 preceding the counter
+                        cnt += 1 << 7
+                    vl += chr(cnt)
+                tosend = bytearray()
+                tosend.extend(map(ord, vl))
+                message = CanMessage(id=addr, interface=CanMessage.CANTYPE_RADAR, data=tosend)
+                self.can_pub.publish(message)
+        self.frame += 1.
+
+    def on_can_message(self, can_msg):
+        if can_msg.interface == CanMessage.CANTYPE_RADAR:
+            if 528 <= can_msg.id <= 559:
```

Comment:
Would be best to move these indexes to class constants. They are duplicated in several places.

Reply:
I'll put a FIXME in there for now to take care of the aesthetic part later once stuff works. The radar IDs are only used here.

Reply:
Actually i just tried doing this and honestly it becomes more confusing. The stuff thats coming off the wire is gonna be the ID. I would rather have the ID just hardcoded right there so it avoids me having to do a mental lookup of a constant.